express-notes-api/index.js

```diff
@@ -0,0 +1,198 @@
+                }
+            } catch (err) {
+                console.log("Error parsing JSON string: ", err);
+            } 
+            
+        })
+    }
+});
+
+// Clients can POST a new note.
+
+app.use(express.json())
+
+app.post('/api/notes', (req, res) => {
+    if (req.body.content === undefined) {
+        res.status(400);
+        res.json({"error": "content is a required field"})
+    } else {
+        fs.readFile(inputFileName, 'utf8', (err, data) => {
+            if (err) {
+                console.log("Error reading File:",  err);
+            } try {
+                const content = JSON.parse(data);
+                const writeObj = {
+                    "content": req.body.content,
+                    "id": content.nextId
+                }
+                let nextId = content.nextId
+                content.notes[nextId] = writeObj;
+                content.nextId++;
+                data = JSON.stringify(content, null, 2);
+                fs.writeFile(inputFileName, data, (err) => {
+                    if (err) {
+                        res.status(500);
+                        res.json({
+                            "error": "An unexpected error occurred."
+                        })
+                    } else {
+                        res.status(201);
+                        res.json(writeObj);
+                    }
+                })
+            } catch (err) {
+                console.log("Error parsing JSON string: ", err);
+            } 
+        })
+    }
+})
+
+
+// Clients can DELETE a note by id.
+
+app.delete('/api/notes/:id', (req, res) => {
+    if (parseInt(req.params.id) <= 0 &&  typeof req.params.id !== 'number' && req.params.id % 1 !== 0) {
+        res.status(400);
+        res.json({"error": "id must be a positive number"})
+    } else {
+        fs.readFile(inputFileName, 'utf8', (err, data) => {
+            if (err) {
+                console.log("Error reading File:",  err);
+            } try {
+                let content = JSON.parse(data);
+                let notesValue ;
+                for (const outObj in content.notes) {
+                    if (content.notes[outObj].id === parseInt(req.params.id)) {
+                        notesValue = content.notes[outObj];
+                        delete content.notes[outObj];
+                    }
+                }
+                content = JSON.stringify(content, null, 2);
+                fs.writeFile(inputFileName, content, (err) => {
+                    if (err) {
+                        res.status(500);
+                        res.json({
+                            "error": "An unexpected error occurred."
+                        })
+                    } else if (notesValue === undefined) {
+                        res.status(404);
+                        res.json({
+                            "error": `note with id ${req.params.id} does not exist`
+                        })
+                    } else {

```

Comment:
If the note with the given `id` doesn't exist, it'd be better to avoid attempting to re-write the file, since nothing has changed.

Reply:
Change the code to avoid re-writing the data file if the updated ID is not found.